app/src/main/java/space/taran/arknavigator/mvp/model/repo/index/ResourceMeta.kt

```diff
@@ -2,21 +2,25 @@ package space.taran.arknavigator.mvp.model.repo.index
 
 import space.taran.arknavigator.mvp.model.dao.ResourceWithExtra
 import space.taran.arknavigator.mvp.model.repo.kind.GeneralKindFactory
+import space.taran.arknavigator.mvp.model.repo.kind.MetaExtraTag
 import space.taran.arknavigator.mvp.model.repo.kind.ResourceKind
+import space.taran.arknavigator.mvp.model.repo.meta.MetadataStorage
 import space.taran.arknavigator.utils.MetaResult
 import space.taran.arknavigator.utils.extension
 import java.io.IOException
 import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.attribute.FileTime
 
+
 data class ResourceMeta(
     val id: ResourceId,
     val name: String,
     val extension: String,
     val modified: FileTime,
     val size: Long,
-    val kind: ResourceKind?
+    val kind: ResourceKind?,
+    val extras: Map<MetaExtraTag, String>?
```

Comment:
We've just realized with Dinar that this slightly wrong, because we already have sub-structures in `ResourceKind` class. Yes, the name is slightly wrong, because `ResourceKind` is not plain enumeration type, but contains sub-structures inside each branch. Is it possible to re-design this PR in order to remove `extras` and use `ResourceKind` internal data instead?

Reply:
NP, I'll make the required changes.

Reply:
Awesome, thanks :+1: